dotnet/src/SemanticKernel.Core/Functions/KernelPluginFactory.cs

```diff
@@ -101,4 +101,18 @@ public static KernelPlugin CreateFromObject(object target, string? pluginName =
     /// <exception cref="ArgumentException"><paramref name="functions"/> contains two functions with the same name.</exception>
     public static KernelPlugin CreateFromFunctions(string pluginName, string? description = null, IEnumerable<KernelFunction>? functions = null) =>
         new DefaultKernelPlugin(pluginName, description, functions);
+
+    /// <summary>Creates a name for a plugin based on its type name.</summary>
+    private static string CreatePluginName(Type type)
+    {
+        string name = type.Name;
+
+        int genericParamIndex = name.IndexOf('`');
+        if (genericParamIndex >= 0)
+        {
+            name = name.Substring(0, genericParamIndex);
+        }
+
+        return KernelFunctionFromMethod.SanitizeMetadataName(name);
```

Comment:
Do we need to include generic type name in plugin name, so LLM can work with same plugin, but different generic type?
```csharp
var builder = Kernel.CreateBuilder();

builder.Plugins.AddFromType<GenericPlugin<int>>();
builder.Plugins.AddFromType<GenericPlugin<string>>();

var kernel = builder.Build();
```

Reply:
I guess it will work with custom plugin name, so maybe we can add support for it when using `AddFromType`. 
```csharp
var builder = Kernel.CreateBuilder();

var intPlugin = new GenericPlugin<int>();
var stringPlugin = new GenericPlugin<string>();

builder.Plugins.AddFromObject(intPlugin, "GenericPluginInt");
builder.Plugins.AddFromObject(stringPlugin, "GenericPluginString");

var kernel = builder.Build();
```

Reply:
Updated how the name is derived.